### ISSUE TITLE
Alarm advising that glycemic line is flat, and finger blood test can be done positively

### ISSUE THREAD

**Issue by ANDREAPIPPI** [NightscoutFoundation/xDrip#1006]
Hello.
I'm using Freestyle Libre with Miaomiao and xDrip +, and like many other people, I suppose, I do two blood tests every day, one in the morning before breakfast, say at 8:00 am, and one before going to sleep, about at 1:00 AM
The problem is that most of the times in those moments the glycemic line is NOT flat, but rises or falls. 
At this point, I do nevertheless the check, and for this reason forecasts are rarely very close to reality, or, sometimes, I wait for the right moment to calibrate, when the line is flat, and often it takes me a long time, obliged to continuously monitor the screen to see the curve's behaviour.
My request: I think it could be interesting to set an alarm that aadvise you when the glycemic value is flat, for example for the last 20 minutes, (let's say + or - 2 mg for the last 4 values ​​compared with the first one) so that at that point it is worth doing the blood test?
I hope this will interest one of the xDrip+ programmers.
Thank you for your time and all that you are doing, and beg your pardon for my bad English.
Bye
Pippo     :-)

**Comment by jsPl** [comment]
Isn't automatic calibration already handling it? It is supposed to "calibrate using new blood glucose readings if the conditions appear right to do so without asking confirmation" (this is actually a quote from the app settings). Not excactly sure what the "right conditions" excatly are though...

**Comment by ANDREAPIPPI** [comment]
Hi jsPI.
Serely you are right, but I cannot find that in my "Settings". 
I'm looking at "Alarms and advices" -->  "Calibration advice", but there I find:
- Alert when a calibration is requested.
- Hours between applications.
- Advice of  requested calibration.
- Override silent mode.
- Even when charging.
- Repeat alerts.
- Alert repeat minutes.
- Initial alert.
but no sign of the text you speak about.
Am I checking in the wrong place?
Or may be my release 190518 is older than your one and doesn't still have that voice?
Thank you for your help.
Bye.
Pippo      :-)

**Comment by jsPl** [comment]
I have the same release. The setting I mention is under:
Less common settings -> Advanced Calibration: Automatic Calibration (this checkbox should be checked by default though)

**Comment by ANDREAPIPPI** [comment]
You are right, I have it and it's checked ON, but it never advised me that it was the right time to take a blood glucose check!
But I see that this feature is "experimental", so may be it doesn't work, yet.
Does it work for you?
Bye and thank you.
Pippo    :-)

**Comment by Navid200** [comment]
@ANDREAPIPPI No one can predict the trend better than you.  Even if the trend is flat right now, it does not mean that this is the right time to calibrate if you have just taken insulin or carb or if you have just started physical activity.

**Comment by ANDREAPIPPI** [comment]
Correct, but I believe that, like me, none has the time to look so frequently his xDrip+ to check if it's the right time to calibrate or not.
But for sure, if xDrip+ could give us an alert when glicemy is flat for more than 15 minutes, we'd be able to evaluate  if it's the case or not to calibrate depending on time distance from insulin or carbs.
Surely it could be far better than calibrating at scheduled times/intervals, believe me.
I don't know if adding this feature is a big / long job, but in many forums this is one of the issues that diabetics talk about.
Bye
Andrea   :-)

**Comment by Navid200** [comment]
I also don't know how long it will take to add this.  But, it's not just adding it.  It will need to be maintained.  It should also not interfere with anything else the code does.  
There are so many requests.  They cannot all be implemented or else, we will just be adding and removing features.  

Someone says let's move this button a little bit up because it's too close to another button and sometimes when I am sleepy, I have hit that instead and don't want to hit that again.  So, we move the button.  Then, someone else says why did you move that button.  it was where it was supposed to be.  But, now it is too far from that other button.  There is no end to this if we just ask how hard is it to implement.  There is a lot more that has to be considered.

Your point brings up a valid concern.  Some of what I have said was meant to offer you a work-around, not to justify why we don't have this.
If you look at the statistics (top left menu) graph, you can see how your variations are at different times of the day.  You can identify times where your glucose is in rapid change, and you can identify times when your glucose varies less.
That can give you an idea what time of day may be best for you to do calibrations.

**Comment by ANDREAPIPPI** [comment]
Hi Navid200.
Yes, I understand your concern and arguments, although the "move the buutton..." example doesn't fit this case, because this would ultimately not be a graphic change, than can be nice for some and ugly for someone else, but a switchable option, which may or may not be chosen.
Anyway, I understand the developers issues, so I close this request.
And thank you for all your work to reduce the absurd number of still open issues.
Bye.
Andrea   :-)

**Comment by Navid200** [comment]
@ANDREAPIPPI 
If you enable Automatic Calibration and you enter all your blood glucose measurements into xDrip, xDrip will use those that fall on a flat trend.

The issue with what you are suggesting is that just because you have a flat trend up to now does not guarantee that this is a good time for calibration.
For example, if you are about to stop an 800-meter run, even if your trend is flat now, it will take a shar drop in a few minutes.
Similarly, if you have some ice cream, your blood glucose will take a sharp rise even if you have had a flat trend up to now.

That's what I mean when I say no one knows the best time for calibration other than you.
xDrip is only aware of the past.  it has no idea about the future; you do because you know what you intend to do next.

Even if a developer develops what you are asking for, I personally at least will not be in favor of merging it because of what I described.
I would like to know what you think.

**Comment by Navid200** [comment]
@tolot27 I just realized that @ANDREAPIPPI closed the issue but you reopened it.
So, would you please answer the question I have asked above?
Thanks

**Comment by tolot27** [comment]
No, you've reoppend the issue at 20th, February. I've just added the enhancement label two days before.

**Comment by nightscout-ricsi** [comment]
I like to have a notification like this. I would like to set:
- what time range the alarm works (do not wake me up by night)
- alarm sound song/file and loudness
- how long (in minutes) the BS must be constant
- what is the BS value range, that means constant (ex. 0.0, 0.1 or 0.2 mml/l)
for conveniuence it would use the BS value rerpesented on the graph.
Why? I might have a good calibration at night, but I don't want to wake up that time. So it would be nice to have a notification, that indicate that one can calibrate right now.

**Comment by Navid200** [comment]
I'm going to convert this to a discussion.
It will still be found by search and it can still be seen by developers.

Thanks
